autometa/config/databases.py

```diff
@@ -66,230 +97,504 @@ class Databases:
             self.config.set('ncbi', option, outfpath)
+        if self.dryrun:
+            return
+        taxdump_md5 = f'{taxdump_fpath}.md5'
+        write_checksum(taxdump_fpath, taxdump_md5)
+        os.remove(taxdump_fpath)
 
-    def update_ncbi(self, options):
-        """Update NCBI database files (taxdump.tar.gz and nr.gz).
+    def download_ncbi_files(self, options):
+        """Download NCBI database files.
 
         Parameters
         ----------
-        options : set
-            Set of options to update
+        options : iterable
+            iterable containing options in 'ncbi' section to download.
 
         Returns
         -------
         NoneType
-            Description of returned object.
+            Will update provided `options` in `self.config`.
 
         Raises
         -------
         ConnectionError
             NCBI file download failed.
 
         """
-        # Download required NCBI database files
-        if not os.path.exists(self.ncbi_dir):
-            os.makedirs(self.ncbi_dir)
-        host = DEFAULT_CONFIG.get('ncbi', 'host')
         for option in options:
-            ftp_fullpath = DEFAULT_CONFIG.get('database_urls', option)
+            host = self.config.get('ncbi', 'host')
+            ftp_fullpath = self.config.get('database_urls', option)
             ftp_fpath = ftp_fullpath.split(host)[-1]
-            if self.config.has_option('ncbi', option):
+
+            if self.config.has_option('ncbi', option) and self.config.get('ncbi', option) is not None:
                 outfpath = self.config.get('ncbi', option)
             else:
                 outfname = os.path.basename(ftp_fpath)
                 outfpath = os.path.join(self.ncbi_dir, outfname)
-            if not self.dryrun and not os.path.exists(outfpath):
-                with FTP(host) as ftp, open(outfpath, 'wb') as fp:
-                    ftp.login()
-                    logger.debug(f'starting {option} download')
-                    result = ftp.retrbinary(f'RETR {ftp_fpath}', fp.write)
-                    if not result.startswith('226 Transfer complete'):
-                        raise ConnectionError(f'{option} download failed')
-                    ftp.quit()
+
             logger.debug(f'UPDATE: (ncbi,{option}): {outfpath}')
             self.config.set('ncbi', option, outfpath)
-        # Extract/format respective NCBI files
-        self.extract_taxdump()
-        self.format_nr()
 
-    def update_markers(self, options):
-        """Update single-copy markers hmms and cutoffs.
+            if self.dryrun:
+                return
+
+            with FTP(host) as ftp, open(outfpath, 'wb') as fp:
+                ftp.login()
+                logger.debug(f'starting {option} download')
+                result = ftp.retrbinary(f'RETR {ftp_fpath}', fp.write)
+                if not result.startswith('226 Transfer complete'):
+                    raise ConnectionError(f'{option} download failed')
+                ftp.quit()
```

Comment:
Here after download, it would make sense to then calculate the md5 of the file and check it against the corresponding md5 URL on the FTP server. Writing the md5 to a file then could signify to the pipeline that the download was successful, and then checking the file later against the md5 file would assure that the file is not corrupted.

Reply:
I saw the addition of `write_checksum` - this is not quite right yet, because if the download was corrupted then the checksum will probably match but be wrong. So we should additionally check that the written checksum file matches the remote checksum at this point.